### DISCUSSION TITLE
Empty note with `tp.file.include`

### DISCUSSION THREAD

**Discussion by tuxpeople** [SilentVoid13/Templater#805] [Help] [answered]
That's a follow up from #741, but I decided to open a new discussion as it is a "new problem".

I have a folder template which only contains java script. It's main purpose is to detect and ask things and to include another template based on its findings.

Full script is in the other discussion, but this is the end of it, where the include happens:

```javascript
if (!(await this.app.vault.adapter.exists(workdir))) {
    await this.app.vault.createFolder(workdir);
}
await tp.file.move(workdir + title);  
await tp.file.include(`[[Templates/work/fragment-${type}]]`);
```

`workdir` is the path where the note should be
`title` is the title of the note
`type` is the type of note it is, which also decides which template to use

My problem is, there's now error but also no content of the new note. It just looks like the last line never happens.

What I tried so far:
- The link is valid, if I print it into the note, I can click on it and will end up at the template
- I also tried the link with the title only, without path. Works as a link, but not as include 
- It also doesn't work when trying it without any variable: `await tp.file.include("[[Templates/work/fragment-Projects]]");`

Has anyone an idea?

**Comment by welpdx** [comment, marked as answer]
`tp.file.include()` doesnt require a path. Just the internal link with the title is ok. For example, if the template that you want to get the content of and added to current file has the name "fragment-Projects", just do `await tp.file.include("[[fragment-Projects]]")`. See [doc](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfileincludeinclude_link-string--tfile). 

If you want to display the content of "[[fragment-Projects]]", you can do that either:
1) in a `<% %>` tag, i.e., `<% await tp.file.include("[[fragment-Projects]]") %>`. 
2) Or you can just add `tR+= await tp.file.include("[[fragment-Projects]]")` to your above script. 
```js
if (!(await this.app.vault.adapter.exists(workdir))) {
    await this.app.vault.createFolder(workdir);
}
await tp.file.move(workdir + title);  
tR+= await tp.file.include(`[[fragment-${type}]]`);
```

**Reply by tuxpeople** [reply]
Thank you very much!

My fault was to assume that `await tp.file.include(`[[fragment-${type}]]`);` would include the content directly. I now moved it below the existing JavaScript code and put it into `<% %>`.

**Reply by welpdx** [reply]
Nice!
